src/main/java/hudson/plugins/scm/koji/model/RPM.java

```diff
@@ -62,11 +63,42 @@ public String getArch() {
 
     @Override
     public String toString() {
-        return nvr + '.' + arch + ".suffix";
+        Suffix suffix = Suffix.INSTANCE;
+        return nvr + '.' + arch + "." + suffix.getSuffix(url);
     }
 
     public String getFilename(String suffix) {
         return Optional.ofNullable(filename).orElse(nvr + '.' + arch + "." + suffix);
     }
 
+    public String getUrl() {
+        return url;
+    }
+
+    public void setUrl(String url) {
+        this.url = url;
+    }
+
+    public boolean hasUrl() {
+        return url != null;
+    }
+
+    public static enum Suffix {
+        INSTANCE;
```

Comment:
Do we really need singleton for this? Suffixes can be `public static final List<String>` and `getSuffix` can be simple private method in `RPM` class.

Reply:
Yes, I belive quite more logic fill be hidden in that class at the end.

Reply:
what logic?